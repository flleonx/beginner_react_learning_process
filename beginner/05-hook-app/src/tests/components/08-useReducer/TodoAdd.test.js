import React from "react";
import { shallow } from "enzyme";
import { TodoAdd } from "../../../components/08-useReducer/TodoAdd";

describe('Tests on <TodoAdd />', () => {

  const handleAddTodo = jest.fn();

  const wrapper = shallow(
    <TodoAdd
      handleAddTodo={ handleAddTodo }
      />
  );

  test('It should show <TodoAdd /> properly', () => {

    expect( wrapper ).toMatchSnapshot();

  });

  test('It should not call handleAddTodo', () => {

    const formSubmit = wrapper.find('form').prop('onSubmit');

    formSubmit({ preventDefault(){} });

    expect( handleAddTodo ).toHaveBeenCalledTimes(0);

  });

  test('It should call handleAddTodo', () => {

    const value = "Learn React";


    wrapper.find('input').simulate('change', {
      target: { value, name: 'description' }
    });

    const formSubmit = wrapper.find('form').prop('onSubmit');

    formSubmit({ preventDefault(){} });

    expect( handleAddTodo ).toHaveBeenCalledTimes(1);
    expect( handleAddTodo ).toHaveBeenCalledWith({
      id: expect.any(Number),
      desc: value,
      done: false
    });

    expect( wrapper.find('input').prop('value') ).toBe('');

  });

});
