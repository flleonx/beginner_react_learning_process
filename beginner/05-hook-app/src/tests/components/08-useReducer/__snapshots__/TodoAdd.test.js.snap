// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <TodoAdd /> It should show <TodoAdd /> properly 1`] = `
<Fragment>
  <h4>
    Add TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Learn ..."
      type="text"
      value=""
    />
    <div
      className="d-grid gap-2"
    >
      <button
        className="btn btn-outline-primary mt-1"
        type="submit"
      >
        Add
      </button>
    </div>
  </form>
</Fragment>
`;
